webapi/src/it/scala/org/knora/webapi/e2e/v1/SipiV1ITSpec.scala

```diff
@@ -175,63 +128,225 @@ class SipiV1ITSpec extends ITSpec(SipiV1ITSpec.config) with TriplestoreJsonProto
+                )
+            )
 
-            // wait for the Future to complete
-            val IIIFPath = Await.result(IIIFPathFuture, 5.seconds)
+            // Send the image in a PUT request to the Knora API server.
+            val knoraPutRequest = Put(baseApiUrl + "/v1/filevalue/" + URLEncoder.encode(firstPageIri.get, "UTF-8"), formData) ~> addCredentials(BasicHttpCredentials(username, password))
+            checkResponseOK(knoraPutRequest)
+        }
 
-            // TODO: now we could try to request the path from Sipi
-            // TODO: we should run Sipi in test mode so it does not do run the preflight request
+        "create an 'incunabula:page' with parameters" in {
+            // The image to be uploaded.
+            val fileToSend = new File(pathToChlaus)
+            assert(fileToSend.exists(), s"File $pathToChlaus does not exist")
 
-            //println(IIIFPath)
+            // A multipart/form-data request containing the image.
+            val sipiFormData = Multipart.FormData(
+                Multipart.FormData.BodyPart(
+                    "file",
+                    HttpEntity.fromPath(MediaTypes.`image/jpeg`, fileToSend.toPath),
+                    Map("filename" -> fileToSend.getName)
+                )
+            )
 
+            // Send a POST request to Sipi, asking it to make a thumbnail of the image.
+            val sipiRequest = Post(baseSipiUrl + "/make_thumbnail", sipiFormData) ~> addCredentials(BasicHttpCredentials(username, password))
+            val sipiResponseJson = getResponseJson(sipiRequest)
+
+            // Request the thumbnail from Sipi.
+            val jsonFields = sipiResponseJson.fields
+            val previewUrl = jsonFields("preview_path").asInstanceOf[JsString].value
+            val sipiGetRequest = Get(previewUrl) ~> addCredentials(BasicHttpCredentials(username, password))
+            checkResponseOK(sipiGetRequest)
+
+            val knoraParams =
+                s"""
+                  |{
+                  |    "restype_id": "http://www.knora.org/ontology/incunabula#page",
+                  |    "properties": {
+                  |        "http://www.knora.org/ontology/incunabula#pagenum": [
+                  |            {"richtext_value": {"utf8str": "test page"}}
+                  |        ],
+                  |        "http://www.knora.org/ontology/incunabula#origname": [
+                  |            {"richtext_value": {"utf8str": "Chlaus"}}
+                  |        ],
+                  |        "http://www.knora.org/ontology/incunabula#partOf": [
+                  |            {"link_value": "http://data.knora.org/5e77e98d2603"}
+                  |        ],
+                  |        "http://www.knora.org/ontology/incunabula#seqnum": [{"int_value": 99999999}]
+                  |    },
+                  |    "file": {
+                  |        "originalFilename" : "${jsonFields("original_filename").asInstanceOf[JsString].value}",
```

Comment:
You can do this rather stylistic improvement if you think it makes sense and then merge.